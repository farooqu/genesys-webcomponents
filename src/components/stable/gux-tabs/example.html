<h1>gux-tabs</h1>

<h2>Example 1 -- tabs left alignment scroll</h2>

<gux-tabs class="example" active-tab="1-2">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="1-1">
      <span>Tab Header 1</span>
    </gux-tab>
    <gux-tab tab-id="1-2">
      <span>Tab Header 2</span>
    </gux-tab>
    <gux-tab tab-id="1-3">
      <span>Tab Header 3 long long long</span>
    </gux-tab>
    <gux-tab tab-id="1-4">
      <span
        >Tab Header 4 really long long long long long long long long long long
        long long long</span
      >
    </gux-tab>
    <gux-tab tab-id="1-5">
      <span
        >Tab Header 5
        longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong</span
      >
    </gux-tab>
    <gux-tab tab-id="1-6">
      <span>Tab Header 6</span>
    </gux-tab>
    <gux-tab tab-id="1-7">
      <span>Tab Header 7</span>
    </gux-tab>
    <gux-tab tab-id="1-8">
      <span>Tab Header 8</span>
    </gux-tab>
    <gux-tab tab-id="1-9">
      <span>Tab Header 9</span>
    </gux-tab>
    <gux-tab tab-id="1-10">
      <span>Tab Header 10</span>
    </gux-tab>
  </gux-tab-list>
  <gux-tab-panel tab-id="1-1">
    <span>Tab content 1</span>
    <div>The current time is: <span id="currentTime"></span></div>
    <div>
      The current selected panel tab-id is: <span id="currenttab-id"></span>
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-2">
    <span>Tab content 2</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-3">
    <span>Tab content 3</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-4">
    <span>Tab content 4</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-5">
    <span>Tab content 5</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-6">
    <span>Tab content 6</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-7">
    <span>Tab content 7</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-8">
    <span>Tab content 8</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-9">
    <span>Tab content 9</span>
  </gux-tab-panel>
  <gux-tab-panel tab-id="1-10">Tab content 10</gux-tab-panel>
</gux-tabs>

<h2>Example 2 -- tabs left alignment</h2>

<gux-tabs class="example">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="2-1"><span>Tab Header 1</span></gux-tab>
    <gux-tab tab-id="2-2"><span>Tab Header 2</span></gux-tab>
    <gux-tab tab-id="2-3"><span>Tab Header 3</span></gux-tab>
    <gux-tab gux-disabled tab-id="2-4"><span>Tab Header 4</span></gux-tab>
    <gux-tab gux-disabled tab-id="2-5"
      ><span>Tab Header 5 long long long</span></gux-tab
    >
  </gux-tab-list>
  <gux-tab-panel tab-id="2-1">Tab content 1</gux-tab-panel>
  <gux-tab-panel tab-id="2-2">Tab content 2</gux-tab-panel>
  <gux-tab-panel tab-id="2-3">Tab content 3</gux-tab-panel>
  <gux-tab-panel tab-id="2-4">Tab content 4</gux-tab-panel>
  <gux-tab-panel tab-id="2-5">Tab content 5</gux-tab-panel>
</gux-tabs>

<h2>Example 3 -- tabs left alignment in a resizable container</h2>

<div
  style="width: 300px; overflow: auto; resize: both; border: 1px dotted black"
>
  <gux-tabs class="example">
    <gux-tab-list slot="tab-list">
      <gux-tab tab-id="3-1"><span>Tab Header 1</span></gux-tab>
      <gux-tab tab-id="3-2"><span>Tab Header 2</span></gux-tab>
      <gux-tab tab-id="3-3"><span>Tab Header 3</span></gux-tab>
      <gux-tab gux-disabled tab-id="3-4"><span>Tab Header 4</span></gux-tab>
      <gux-tab gux-disabled tab-id="3-5"
        ><span>Tab Header 5 long long long</span></gux-tab
      >
    </gux-tab-list>
    <gux-tab-panel tab-id="3-1">Tab content 1</gux-tab-panel>
    <gux-tab-panel tab-id="3-2">Tab content 2</gux-tab-panel>
    <gux-tab-panel tab-id="3-3">Tab content 3</gux-tab-panel>
    <gux-tab-panel tab-id="3-4">Tab content 4</gux-tab-panel>
    <gux-tab-panel tab-id="3-5">Tab content 5</gux-tab-panel>
  </gux-tabs>
</div>

<h2>Example 4 -- tabs vertical alignment in a resizable container</h2>

<div
  style="height: 300px; overflow: auto; resize: both; border: 1px dotted black"
>
  <gux-tabs orientation="vertical" class="example">
    <gux-tab-list slot="tab-list">
      <gux-tab tab-id="4-1">
        <span>Tab Header 1</span>
      </gux-tab>
      <gux-tab tab-id="4-2">
        <span>Tab Header 2</span>
      </gux-tab>
      <gux-tab gux-disabled tab-id="4-3"> <span>Tab Header 3</span></gux-tab>
      <gux-tab tab-id="4-4">
        <span>Tab Header long long long</span>
      </gux-tab>
      <gux-tab tab-id="4-5"><span>Tab Header 5</span></gux-tab>
    </gux-tab-list>
    <gux-tab-panel tab-id="4-1">Tab content 1</gux-tab-panel>
    <gux-tab-panel tab-id="4-2">Tab content 2</gux-tab-panel>
    <gux-tab-panel tab-id="4-3">Tab content 3</gux-tab-panel>
    <gux-tab-panel tab-id="4-4">Tab content 4</gux-tab-panel>
    <gux-tab-panel tab-id="4-5">Tab content 5</gux-tab-panel>
  </gux-tabs>
</div>

<h2>Example 5 -- tabs icons only</h2>

<gux-tabs class="example">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="5-1">
      <gux-icon icon-name="agent" screenreader-text="Agent"></gux-icon>
    </gux-tab>
    <gux-tab tab-id="5-2">
      <gux-icon
        icon-name="user-directory"
        screenreader-text="User Directory"
      ></gux-icon>
    </gux-tab>
    <gux-tab tab-id="5-3">
      <gux-icon icon-name="cobrowse" screenreader-text="Co-browse"></gux-icon>
    </gux-tab>
    <gux-tab tab-id="5-4">
      <gux-icon icon-name="user-group" screenreader-text="Users"></gux-icon>
    </gux-tab>
    <gux-tab tab-id="5-5">
      <gux-icon
        icon-name="attachment"
        screenreader-text="Attachments"
      ></gux-icon>
    </gux-tab>
  </gux-tab-list>
  <gux-tab-panel tab-id="5-1"> Tab content 1 </gux-tab-panel>
  <gux-tab-panel tab-id="5-2"> Tab content 2 </gux-tab-panel>
  <gux-tab-panel tab-id="5-3"> Tab content 3 </gux-tab-panel>
  <gux-tab-panel tab-id="5-4"> Tab content 4 </gux-tab-panel>
  <gux-tab-panel tab-id="5-5"> Tab content 5 </gux-tab-panel>
</gux-tabs>

<h2>Example 6 -- tabs icons and title</h2>

<gux-tabs class="example">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="6-1">
      <gux-icon icon-name="agent" decorative="true"></gux-icon>
      <span>Tab Header 1</span>
    </gux-tab>
    <gux-tab tab-id="6-2">
      <gux-icon icon-name="user-directory" decorative="true"></gux-icon>
      <span>Tab Header 2</span>
    </gux-tab>
    <gux-tab tab-id="6-3">
      <gux-icon icon-name="cobrowse" decorative="true"></gux-icon>
      <span>Tab Header 3 long long long long</span>
    </gux-tab>
    <gux-tab tab-id="6-4">
      <gux-icon icon-name="user-group" decorative="true"></gux-icon>
      <span>Tab Header 4</span>
    </gux-tab>
    <gux-tab tab-id="6-5">
      <gux-icon icon-name="attachment" decorative="true"></gux-icon>
      <span>Tab Header 5</span>
    </gux-tab>
  </gux-tab-list>
  <gux-tab-panel tab-id="6-1"> Tab content 1 </gux-tab-panel>
  <gux-tab-panel tab-id="6-2"> Tab content 2 </gux-tab-panel>
  <gux-tab-panel tab-id="6-3"> Tab content 3 </gux-tab-panel>
  <gux-tab-panel tab-id="6-4"> Tab content 4 </gux-tab-panel>
  <gux-tab-panel tab-id="6-5"> Tab content 5 </gux-tab-panel>
</gux-tabs>

<h2>Example 7 -- tabs full width</h2>

<gux-tabs alignment="full-width" class="example">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="7-1">
      <span>Tab Header 1</span>
    </gux-tab>
    <gux-tab tab-id="7-2">
      <span>Tab Header 2</span>
    </gux-tab>
    <gux-tab tab-id="7-3">
      <span>Tab Header 3</span>
    </gux-tab>
  </gux-tab-list>
  <gux-tab-panel tab-id="7-1"> Tab content 1 </gux-tab-panel>
  <gux-tab-panel tab-id="7-2"> Tab content 2 </gux-tab-panel>
  <gux-tab-panel tab-id="7-3"> Tab content 3 </gux-tab-panel>
</gux-tabs>

<h2>Example 8 -- tabs center aligned</h2>

<gux-tabs alignment="center" class="example">
  <gux-tab-list slot="tab-list">
    <gux-tab tab-id="8-1">
      <span>Tab Header 1</span>
    </gux-tab>
    <gux-tab tab-id="8-2">
      <span>Tab Header 2</span>
    </gux-tab>
    <gux-tab tab-id="8-3">
      <span>Tab Header 3</span>
    </gux-tab>
  </gux-tab-list>
  <gux-tab-panel tab-id="8-1"> Tab content 1 </gux-tab-panel>
  <gux-tab-panel tab-id="8-2"> Tab content 2 </gux-tab-panel>
  <gux-tab-panel tab-id="8-3"> Tab content 3 </gux-tab-panel>
</gux-tabs>

<style
  onload="(function () {
    document.querySelector(`gux-tab-panel[tab-id='1-1']`).addEventListener('guxactivepanelchange', 
    (event) => {
      const currentTime = new Date().toLocaleTimeString();
      document.querySelector('#currentTime').innerHTML = currentTime;
      document.querySelector('#currenttab-id').innerHTML = event.detail;
    })
  }
  )()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
