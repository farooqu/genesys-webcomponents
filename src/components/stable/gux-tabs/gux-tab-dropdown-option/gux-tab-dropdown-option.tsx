import { Component, h, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-tab-dropdown-option.less',
  tag: 'gux-tab-dropdown-option'
})
export class GuxTabDropdownOption {
  /**
   * unique id for the option
   */
  @Prop() optionId: string;

  /**
   * name of the gux-icon to display for the option
   */
  @Prop() iconName: string;

  render() {
    return (
      <button class="tab-dropdown-option">
        <gux-icon icon-name={this.iconName} decorative={true}></gux-icon>
        <span>
          <slot />
        </span>
      </button>
    );
  }
}
