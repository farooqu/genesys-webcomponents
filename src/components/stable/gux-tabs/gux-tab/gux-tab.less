@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/spacing.less';
@import (reference) '../../../../style/mixins.less';

gux-tabs[orientation='vertical'] {
  .gux-tab {
    display: flex;
    justify-content: flex-end;
    padding: @spacing-small @spacing-medium;

    &.gux-active {
      padding-right: @spacing-medium - 2px;
      border-right: 2px solid @gux-blue-60;
    }

    &:hover:not(.gux-active):not(.gux-disabled) {
      padding-right: @spacing-medium - 2px;
      border-right: 2px solid @gux-grey-10;
    }
  }
}

gux-tabs:not([orientation='vertical']) {
  .gux-tab {
    box-sizing: border-box;
    padding-bottom: 2px;

    &.gux-active {
      padding-bottom: 0;
      border-bottom: 2px solid @gux-blue-60;
    }

    &:hover:not(.gux-active):not(.gux-disabled) {
      padding-bottom: 0;
      border-bottom: 2px solid @gux-grey-10;
    }

    gux-tooltip-title {
      margin: auto;
    }
  }
}

gux-tabs[alignment='center'] {
  .gux-scrollable-section {
    justify-content: center;
  }
}

gux-tabs[alignment='full-width'] {
  .gux-scrollable-section {
    flex-grow: 1;
  }

  gux-tab {
    width: 100%;
    max-width: 100%;

    .gux-tab {
      width: 100%;
      max-width: 100%;
    }
  }
}

gux-tab {
  display: flex;
  max-width: 160px;

  .gux-tab {
    display: flex;
    align-items: center;
    width: 100%;
    max-width: 160px;
    height: 40px;
    padding: 0 @spacing-medium;
    cursor: pointer;
    background-color: #fff;
    border: none;

    &.gux-disabled {
      cursor: default;

      gux-tooltip-title {
        cursor: default;
        opacity: 50%;
      }
    }

    &:focus-visible {
      padding: 0 13px 3px 13px;
      border: 3px solid @gux-blue-90;
      border-radius: 5px;
      outline: none;
    }

    gux-tooltip-title {
      max-width: 160px;
      overflow: hidden;
      font-size: 14px;
      color: @gux-black-50;
      text-overflow: ellipsis;
      white-space: nowrap;
      cursor: pointer;
    }
  }
}
