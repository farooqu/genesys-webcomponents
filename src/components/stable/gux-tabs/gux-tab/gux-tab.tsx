import {
  Component,
  Event,
  EventEmitter,
  h,
  JSX,
  Listen,
  Method,
  Prop,
  State
} from '@stencil/core';

@Component({
  styleUrl: 'gux-tab.less',
  tag: 'gux-tab',
  shadow: false
})
export class GuxTab {
  private buttonElement: HTMLButtonElement;

  /**
   * Tab id for the tab
   */
  @Prop()
  tabId: string;

  /**
   * Specifies if tab is disabled
   */
  @Prop()
  guxDisabled: boolean = false;

  /**
   * Specifies if the tab title is just an icon
   */
  @Prop()
  iconOnly: boolean = false;

  @State()
  active: boolean = false;

  @Listen('click')
  onClick() {
    if (!this.active && !this.guxDisabled) {
      this.internalactivatetabpanel.emit(this.tabId);
    }
  }

  @Event()
  internalactivatetabpanel: EventEmitter<string>;

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async guxSetActive(active: boolean): Promise<void> {
    this.active = active;
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async guxFocus(): Promise<void> {
    this.buttonElement.focus();
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async guxGetActive() {
    return this.active;
  }

  render(): JSX.Element {
    return (
      <button
        class={{
          'gux-disabled': this.guxDisabled,
          'gux-tab': true,
          'gux-active': this.active
        }}
        type="button"
        aria-disabled={this.guxDisabled.toString()}
        id={`gux-${this.tabId}-tab`}
        role="tab"
        aria-controls={`gux-${this.tabId}-panel`}
        aria-selected={this.active.toString()}
        tabIndex={this.active ? 0 : -1}
        ref={el => (this.buttonElement = el)}
      >
        <gux-tooltip-title tab-width={122}>
          <slot />
        </gux-tooltip-title>
      </button>
    ) as JSX.Element;
  }
}
