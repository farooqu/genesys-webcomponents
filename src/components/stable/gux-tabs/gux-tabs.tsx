import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  JSX,
  Listen,
  Method,
  Prop,
  State,
  Watch
} from '@stencil/core';

import { trackComponent } from '../../../usage-tracking';

import { GuxTabsOrientation, GuxTabsAlignment } from './gux-tabs-types';

@Component({
  styleUrl: 'gux-tabs.less',
  tag: 'gux-tabs',
  shadow: true
})
export class GuxTabs {
  @Element()
  root: HTMLElement;

  /**
   * tabId of the currently selected tab
   */
  @Prop({ mutable: true })
  activeTab: string;
  /**
   * Specifies horizontal or vertical orientation of tabs
   */
  @Prop()
  orientation: GuxTabsOrientation = 'horizontal';

  /**
   * Specifies left aligned, centered, or full width tabs
   */
  @Prop()
  alignment: GuxTabsAlignment = 'left';

  @State()
  tabList: HTMLGuxTabListElement;

  @State()
  tabPanels: HTMLGuxTabPanelElement[] = [];

  /**
   * Triggers when the active tab changes.
   */
  @Event()
  guxactivetabchange: EventEmitter<string>;

  @Watch('activeTab')
  watchActiveTab(newValue: string) {
    this.guxactivetabchange.emit(newValue);
  }

  @Listen('internalactivatetabpanel')
  onInternalActivateTabPanel(event: CustomEvent): void {
    event.stopPropagation();

    const tabId = event.detail as string;

    this.activateTab(tabId, this.tabList, this.tabPanels);
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async guxActivate(tabId: string): Promise<void> {
    this.activateTab(tabId, this.tabList, this.tabPanels);
  }

  private onSlotchange(): void {
    const [tabListSlot, defaultSlot] = Array.from(
      this.root.shadowRoot.querySelectorAll('slot')
    );

    this.tabList = tabListSlot.assignedElements()[0] as HTMLGuxTabListElement;
    this.tabPanels = defaultSlot.assignedElements() as HTMLGuxTabPanelElement[];

    this.activateTab(this.activeTab, this.tabList, this.tabPanels);
  }

  private activateTab(
    tabId: string,
    tabList: HTMLGuxTabListElement,
    panels: HTMLGuxTabPanelElement[]
  ): void {
    if (tabId) {
      this.activeTab = tabId;
    } else {
      this.activeTab = panels[0].tabId;
    }

    void tabList.guxSetActive(this.activeTab);
    panels.forEach(
      panel => void panel.guxSetActive(panel.tabId === this.activeTab)
    );
  }

  componentWillLoad(): void {
    trackComponent(this.root);
  }

  render(): JSX.Element {
    return (
      <Host>
        <div class={`gux-tabs gux-${this.alignment} gux-${this.orientation}`}>
          <slot name="tab-list"></slot>
          <div class={`gux-${this.alignment} gux-${this.orientation}`}>
            <slot onSlotchange={this.onSlotchange.bind(this)}></slot>
          </div>
        </div>
      </Host>
    ) as JSX.Element;
  }
}
