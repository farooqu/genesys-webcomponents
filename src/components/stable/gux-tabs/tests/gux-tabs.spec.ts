import { newSpecPage, SpecPage } from '@stencil/core/testing';

import { GuxTabs } from '../gux-tabs';
import { GuxTab } from '../gux-tab/gux-tab';
import { GuxTabDropdownOption } from '../gux-tab-dropdown-option/gux-tab-dropdown-option';

const components = [GuxTabs, GuxTab, GuxTabDropdownOption];
const language = 'en';

describe('gux-tabs', () => {
  let page: SpecPage;

  beforeEach(async () => {
    page = await newSpecPage({
      components,
      html: `
        <gux-tabs id="interactive">
          <gux-tab tab-id="1" tab-icon-name="lock">
            <span slot="title"> Hello World </span>
            <span slot="dropdown-options">
              <gux-tab-dropdown-option
                option-id="1"
                icon-name="edit"
                onclick="notify(event)"
              >
                Edit
              </gux-tab-dropdown-option>
              <gux-tab-dropdown-option
                option-id="2"
                icon-name="clone"
                onclick="notify(event)"
              >
                Clone
              </gux-tab-dropdown-option>
              <gux-tab-dropdown-option
                option-id="3"
                icon-name="share"
                onclick="notify(event)"
              >
                Share
              </gux-tab-dropdown-option>
              <gux-tab-dropdown-option
                option-id="4"
                icon-name="download"
                onclick="notify(event)"
              >
                Download
              </gux-tab-dropdown-option>
            </span>
          </gux-tab>

          <gux-tab tab-id="2" tab-icon-name="lock">
            <span slot="title"> Hello World 2 </span>
          </gux-tab>
        </gux-tabs>
      `,
      language
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxTabs);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
