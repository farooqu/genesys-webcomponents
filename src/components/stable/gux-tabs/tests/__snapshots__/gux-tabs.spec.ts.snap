// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-tabs #render should render as expected 1`] = `
<gux-tabs class="example">
  <mock:shadow-root>
    <div class="gux-horizontal gux-left gux-tabs">
      <slot name="tab-list"></slot>
      <div class="gux-horizontal gux-left">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-tab-list slot="tab-list">
    <!---->
    <div class="gux-tab-container">
      <div class="gux-scroll-button-container"></div>
      <div aria-owns="gux-2-1-tab gux-2-2-tab gux-2-3-tab gux-2-4-tab gux-2-5-tab" class="gux-scrollable-section" role="tablist">
        <gux-tab tab-id="2-1">
          <!---->
          <button aria-controls="gux-2-1-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-1-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 1
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab tab-id="2-2">
          <!---->
          <button aria-controls="gux-2-2-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-2-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 2
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab tab-id="2-3">
          <!---->
          <button aria-controls="gux-2-3-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-3-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 3
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab gux-disabled="" tab-id="2-4">
          <!---->
          <button aria-controls="gux-2-4-panel" aria-disabled="true" aria-selected="false" class="gux-disabled gux-tab" id="gux-2-4-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 4
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab gux-disabled="" tab-id="2-5">
          <!---->
          <button aria-controls="gux-2-5-panel" aria-disabled="true" aria-selected="false" class="gux-disabled gux-tab" id="gux-2-5-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 5
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
      </div>
      <div class="gux-scroll-button-container"></div>
    </div>
  </gux-tab-list>
  <gux-tab-panel tab-id="2-1">
    <!---->
    <div aria-labelledby="gux-2-1-tab" hidden="" id="gux-2-1-panel" role="tabpanel" tabindex="0">
      Tab content 1
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-2">
    <!---->
    <div aria-labelledby="gux-2-2-tab" hidden="" id="gux-2-2-panel" role="tabpanel" tabindex="0">
      Tab content 2
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-3">
    <!---->
    <div aria-labelledby="gux-2-3-tab" hidden="" id="gux-2-3-panel" role="tabpanel" tabindex="0">
      Tab content 3
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-4">
    <!---->
    <div aria-labelledby="gux-2-4-tab" hidden="" id="gux-2-4-panel" role="tabpanel" tabindex="0">
      Tab content 4
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-5">
    <!---->
    <div aria-labelledby="gux-2-5-tab" hidden="" id="gux-2-5-panel" role="tabpanel" tabindex="0">
      Tab content 5
    </div>
  </gux-tab-panel>
</gux-tabs>
`;

exports[`gux-tabs #render should render vertical tabs as expected 1`] = `
<gux-tabs orientation="vertical">
  <mock:shadow-root>
    <div class="gux-left gux-tabs gux-vertical">
      <slot name="tab-list"></slot>
      <div class="gux-left gux-vertical">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <gux-tab-list slot="tab-list">
    <!---->
    <div class="gux-tab-container">
      <div class="gux-scroll-button-container"></div>
      <div aria-owns="gux-2-1-tab gux-2-2-tab gux-2-3-tab gux-2-4-tab gux-2-5-tab" class="gux-scrollable-section" role="tablist">
        <gux-tab tab-id="2-1">
          <!---->
          <button aria-controls="gux-2-1-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-1-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 1
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab tab-id="2-2">
          <!---->
          <button aria-controls="gux-2-2-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-2-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 2
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab tab-id="2-3">
          <!---->
          <button aria-controls="gux-2-3-panel" aria-disabled="false" aria-selected="false" class="gux-tab" id="gux-2-3-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 3
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab gux-disabled="" tab-id="2-4">
          <!---->
          <button aria-controls="gux-2-4-panel" aria-disabled="true" aria-selected="false" class="gux-disabled gux-tab" id="gux-2-4-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 4
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
        <gux-tab gux-disabled="" tab-id="2-5">
          <!---->
          <button aria-controls="gux-2-5-panel" aria-disabled="true" aria-selected="false" class="gux-disabled gux-tab" id="gux-2-5-tab" role="tab" tabindex="-1" type="button">
            <gux-tooltip-title tab-width="122">
              <span>
                Tab Header 5
              </span>
            </gux-tooltip-title>
          </button>
        </gux-tab>
      </div>
      <div class="gux-scroll-button-container"></div>
    </div>
  </gux-tab-list>
  <gux-tab-panel tab-id="2-1">
    <!---->
    <div aria-labelledby="gux-2-1-tab" hidden="" id="gux-2-1-panel" role="tabpanel" tabindex="0">
      Tab content 1
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-2">
    <!---->
    <div aria-labelledby="gux-2-2-tab" hidden="" id="gux-2-2-panel" role="tabpanel" tabindex="0">
      Tab content 2
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-3">
    <!---->
    <div aria-labelledby="gux-2-3-tab" hidden="" id="gux-2-3-panel" role="tabpanel" tabindex="0">
      Tab content 3
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-4">
    <!---->
    <div aria-labelledby="gux-2-4-tab" hidden="" id="gux-2-4-panel" role="tabpanel" tabindex="0">
      Tab content 4
    </div>
  </gux-tab-panel>
  <gux-tab-panel tab-id="2-5">
    <!---->
    <div aria-labelledby="gux-2-5-tab" hidden="" id="gux-2-5-panel" role="tabpanel" tabindex="0">
      Tab content 5
    </div>
  </gux-tab-panel>
</gux-tabs>
`;
